auto_submit/lib/service/validation_service.dart

```diff
@@ -146,20 +145,112 @@ class ValidationService {
       }
     }
     // If we got to this point it means we are ready to submit the PR.
-    bool processed = await processMerge(config, result, messagePullRequest);
-    if (processed) await pubsub.acknowledge('auto-submit-queue-sub', ackId);
-    log.info('Ack the processed message : $ackId.');
+    await processMergeSafely(
+        config, gitHubService, messagePullRequest, result, pubsub, ackId, slug, prNumber, Config.kAutosubmitLabel);
+  }
+
+  /// The logic for processing a revert request and opening the follow up
+  /// review issue in github.
+  Future<void> processRevertRequest(
```

Comment:
For large methods, the Dart way is to make everything a named parameter. Generally, after 2 parameters is considered large.

Reply:
Odd why the minute the parameters become named they need to be explicitly defined as nullable...

Reply:
You can do `required String myParam`